rfcs/52-unstable_features.md

```diff
@@ -0,0 +1,89 @@
+- Something in between that would benefit from iterative work? Here come the unstable feature workflow for you!
+
+You will need to open a tracking issue to explain the new feature you want to add. If it looks good, you'll be ready to open that PR with a feature gate for your changes!
+
+## Implementation strategy
+
+The new workflow for this kind of feature would be:
+- Submit an issue explaining the feature you want to add under a feature gate. This will be the tracking issue for stabilization once approved. This issue gets a new label `S-Stabilization`.
+- If approved, submit a PR with the new feature and the feature gate
+- This PR can be merged by someone with the merge rights on the related part of Bevy as soon as it's `S-Ready-For-Final-Review`, without an approval by the main Bevy maintainers.
+- Subsequent PRs on the feature must be linked to the tracking issue and follow the same process.
+- Once the feature is finished, enter the stabilization period. Open a PR that will remove the feature gate. This PR needs to be approved by the main Bevy maintainers.
+
+The feature gate should follow the following convention for its name: `unstable-<#tracking-issue-number>-<feature-name>`.
+
+An existing PR can be retroffited in this process on the suggestion of Bevy community members, and if the PR author agrees.
+
+The RFCs and the unstable features are closely related. An RFC can be implemented as an unstable feature to help resolve issues that need more experimentation, or split a large implementation. An unstable feature could require an RFC to reach stabilization. For example:
+- RFC to Unstable Feature
+    - Open a RFC, reaching a design
+    - Implement part of the RFC as an unstable feature
+    - Feedback into the RFC with the gained knowledge of implementation and usage
+- Unstable Feature to RFC
+    - Open an tracking issue, start implementing it
+    - Part of the unstable feature is larger than expected with several design possibilities
+    - Open a RFC, reaching a decision on the design
+    - Implement the RFC for the unstable feature
+
+
+Each PR will still need to be approved by Bevy community members, and would still need to have the same level of quality as other PRs merged.
+
+The PRs for the initial feature, various changes during its finalization and for the stabilization can be opened by different persons.
+
+The feature gate should not be exposed on the main `bevy` crate, but only on the subcrates where it is relevant. It can be enabled as a user by adding dependencies directly on the subcrates and enabling the feature. The tracking issue should list the subcrates impacted.
+
+For organization and discovery, a GitHub project can be used to track the stabilization issues.
+
+While it should be avoided, if another PR breaks an unstable feature, it is not the PR author responsability to fix it. They may do it if they want, otherwise it must be reported on the unstable feature tracking issue.
+
```

Comment:
Not something to add to this RFC, but it would be useful to come up with some "current" PRs that would benefit from being unstable features.

Reply:
The PR that motivated me to write that RFC is https://github.com/bevyengine/bevy/pull/3884

As you said yourself
> But this feels like a "let people yell at us if we did something wrong" sort of situation.

This RFC generalises that possibility: to have an advanced feature that we want to test, and let the interested people manage until it's ready for common use